documentation/src/docs/streamcuts.md

```diff
@@ -0,0 +1,92 @@
+point in time. The offset always points to the event boundary and hence there will be no offset pointing to
+an incomplete event.
+
+The `StreamCut` representing the tail of the stream (with the newest event) is an ever changing one since
+events can be continuously added to the stream and the `StreamCut` pointing to the tail of the stream with
+newer events would have a different value. Similarly the `StreamCut` representing the head of the
+stream (with the oldest event) is an ever changing one as the stream retention policy could truncate the stream
+and the `StreamCut` pointing to the head of the stream post truncation would have a different value.
+```StreamCut.UNBOUNDED``` is used to represent such a position in the stream and the user can use it to
+specify this ever changing stream position (both head and tail of the stream).
+
+It should be noted that `StreamCut`s obtained using the streaming client and batch client can be used
+interchangeably.
+
+## StreamCut with Streaming clients
+
+A ReaderGroup is a named collection of Readers that together, in parallel, read Events from a given Stream. Every
+Reader is always associated with a ReaderGroup. `StreamCut`(s) can be obtained from a ReaderGroup using the
+following api ```io.pravega.client.stream.ReaderGroup.getStreamCuts ```. This api returns a
+```Map<Stream, StreamCut>``` which represents the last known position of the Readers for all the streams managed by
+the ReaderGroup.
+
+A `StreamCut` can be used to configure a ReaderGroup to enable bounded processing of a Stream. The start
+and/or end `StreamCut` of a Stream can be passed as part of the ReaderGroup configuration. The below example
+shows the different ways to use `StreamCut`s as part of the ReaderGroup configuration.
+
+```java
+/*
+ * The below ReaderGroup configuration ensures that the readers belonging to
+ * the ReaderGroup read events from
+ *   - Stream "s1" from startStreamCut1 (representing the oldest event) upto
+          endStreamCut1 (representing the newest event)
+ *   - Stream "s2" from startStreamCut2 upto the tail of the stream, this is similar to using StreamCut.UNBOUNDED
+ *        for endStreamCut.
+ *   - Stream "s3" from the current head of the stream upto endStreamCut2
+ *   - Stream "s4" from the current head of the stream upto the tail of the stream.
+ */
+ReaderGroupConfig.builder()
+                .stream("scope/s1", startStreamCut1, endStreamCut1)
+                .stream("scope/s2", startStreamCut2)
+                .stream("scope/s3", StreamCut.UNBOUNDED, endStreamCut2)
+                .stream("scope/s4")
+                .build();
+```
+
+The below API can be used to reset an existing ReaderGroup with a new ReaderGroup configuration instead creating a
+ReaderGroup.
+```
+/*
+ * ReaderGroup api used to reset a ReaderGroup to a newer ReaderGroup configuration.
+ */
+io.pravega.client.stream.ReaderGroup.resetReaderGroup(ReaderGroupConfig config)
+```
+## StreamCut with BatchClient
```

Comment:
I'm not seeing this section talking about bounded processing and so on.

Reply:
The section talks about how user can perform processing given the upper and lower bound of the Stream. Let me know if anything needs to improved/added here.

Reply:
Talk about bounded processing explicitly. This seems to be implicitly assuming that the reader will understand that this can be used for bounded processing. Talk about it explicitly.

Reply:
ok, update the documentation accordingly.